esp-hal/src/exception_handler/mod.rs

```diff
@@ -15,18 +15,57 @@ unsafe extern "C" fn __user_exception(
     );
 }
 
+#[cfg(xtensa)]
+#[unsafe(no_mangle)]
+#[unsafe(link_section = ".rwtext")]
```

Comment:
Does this have to be in RAM? We should be able to load this from flash, and the performance isn't critical as this will just die miserably if called.

Reply:
I thought the same - for now this was just copied, but in this case since we know "the world is about to end" a cache miss is not the worst thing that could happen 👍